File: pascalstriangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res(numRows, vector<int>());
        if (numRows == 0) return res;
        for (int i = 0; i < numRows; i++) {
            if (i == 0) res[i].push_back(1);
            else {
                res[i].push_back(1);
                for (int j = 1; j < res[i-1].size(); j++) {
                    res[i].push_back(res[i-1][j-1]+res[i-1][j]);
                }
                res[i].push_back(1);
            }
        }
        return res;
    }
};

File: python/pascalstriangle.py
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        res = [0 for _ in xrange(rowIndex+1)]
        res[0] = 1
        for i in xrange(1,rowIndex+1):
            for j in xrange(i,0,-1):
                res[j] += res[j-1]
        return res
